renderer/backend/RenderBackend.cpp

```diff
@@ -0,0 +1,256 @@
+	depthStage.Shutdown();
+	drawBatchExecutor.Destroy();
+}
+
+void RenderBackend::DrawView( const viewDef_t *viewDef ) {
+	// we will need to do a new copyTexSubImage of the screen when a SS_POST_PROCESS material is used
+	backEnd.currentRenderCopied = false;
+	backEnd.afterFogRendered = false;
+
+	GL_PROFILE( "DrawView" );
+
+	// skip render bypasses everything that has models, assuming
+	// them to be 3D views, but leaves 2D rendering visible
+	if ( viewDef->viewEntitys && r_skipRender.GetBool() ) {
+		return;
+	}
+
+	// skip render context sets the wgl context to NULL,
+	// which should factor out the API cost, under the assumption
+	// that all gl calls just return if the context isn't valid
+	if ( viewDef->viewEntitys && r_skipRenderContext.GetBool() ) {
+		GLimp_DeactivateContext();
+	}
+	backEnd.pc.c_surfaces += viewDef->numDrawSurfs;
+
+	RB_ShowOverdraw();
+
+
+	int processed;
+
+	backEnd.depthFunc = GLS_DEPTHFUNC_EQUAL;
+
+	// clear the z buffer, set the projection matrix, etc
+	RB_BeginDrawingView();
+
+	backEnd.lightScale = r_lightScale.GetFloat();
+	backEnd.overBright = 1.0f;
+
+	drawSurf_t **drawSurfs = ( drawSurf_t ** )&viewDef->drawSurfs[ 0 ];
+	int numDrawSurfs = viewDef->numDrawSurfs;
+
+	// if we are just doing 2D rendering, no need to fill the depth buffer
+	if ( viewDef->viewEntitys ) {
+		// fill the depth buffer and clear color buffer to black except on subviews
+		depthStage.DrawDepth( viewDef, drawSurfs, numDrawSurfs );
+		if( ambientOcclusion->ShouldEnableForCurrentView() ) {
+			ambientOcclusion->ComputeSSAOFromDepth();
+		}
+		DrawShadowsAndInteractions( viewDef );
+	}
+		
+	// now draw any non-light dependent shading passes
+	int RB_STD_DrawShaderPasses( drawSurf_t **drawSurfs, int numDrawSurfs );
+	processed = RB_STD_DrawShaderPasses( drawSurfs, numDrawSurfs );
+
+	// fog and blend lights
+	void RB_STD_FogAllLights( bool translucent );
+	RB_STD_FogAllLights( false );
+
+	// refresh fog and blend status 
+	backEnd.afterFogRendered = true;
+
+	// now draw any post-processing effects using _currentRender
+	if ( processed < numDrawSurfs ) {
+		RB_STD_DrawShaderPasses( drawSurfs + processed, numDrawSurfs - processed );
+	}
+
+	RB_STD_FogAllLights( true ); // 2.08: second fog pass, translucent only
+
+	RB_RenderDebugTools( drawSurfs, numDrawSurfs );
+
+	// restore the context for 2D drawing if we were stubbing it out
+	if ( r_skipRenderContext.GetBool() && viewDef->viewEntitys ) {
+		GLimp_ActivateContext();
+		RB_SetDefaultGLState();
+	}
+}
+
+void RenderBackend::EndFrame() {
+	drawBatchExecutor.EndFrame();
+	if (GLAD_GL_ARB_bindless_texture) {
+		globalImages->MakeUnusedImagesNonResident();
+	}
+}
+
+bool RenderBackend::ShouldUseBindlessTextures() const {
+	return GLAD_GL_ARB_bindless_texture && r_useBindlessTextures.GetBool();
+}
+
+void RenderBackend::DrawInteractionsWithShadowMapping(viewLight_t *vLight) {
+	void RB_GLSL_DrawInteractions_ShadowMap( const drawSurf_t *surf, bool clear = false );
```

Comment:
Could you please remove default argument here?

Using local function declarations with default arguments can seriously mess things up if default arguments don't match.